### PR TITLE
Media fixes

### PR DESCRIPTION
**Pull request by Pikkupomo**
Fix issue with saving images
Fix issue with product images and medias mixing

Refs SHOOP-1345

### CODE REVIEW

File: shoop/admin/modules/products/views/forms.py [hunk 1]
```diff
@@ -183,6 +184,13 @@ def __init__(self, **kwargs):
             self.fields["kind"].initial = self.allowed_media_kinds[0]
             self.fields["kind"].widget = forms.HiddenInput()
 
+        if self.disallowed_media_kinds:
+            choices = []
+            for (i, choice) in self.fields["kind"].choices:
```
Comment (akx) [inline shoop/admin/modules/products/views/forms.py]: Uh, so what's happening in this loop?

**Comment by akx** [comment]
I don't understand what the issue is ("error when saving image with URL"?) and how this fixes it.

The second commit is equally confusing (and ref-less)...

**Comment by Pikkupomo** [comment]
>I don't understand what the issue is ("error when saving image with URL"?) and how this fixes it.

The original ticket description was misleading, it has been fixed

>The second commit is equally confusing (and ref-less)...

It should't confuse you anymore... I hope :)

File: shoop/admin/modules/products/views/forms.py [hunk 2]
```diff
@@ -178,10 +178,19 @@ def __init__(self, **kwargs):
         self.fields["file"].widget = MediaChoiceWidget()  # Filer misimplemented the field; we need to do this manually.
         self.fields["file"].required = True
 
-        if len(self.allowed_media_kinds) == 1:
-            # only one media kind given, no point showing the dropdown
+        if self.allowed_media_kinds:
+            if len(self.allowed_media_kinds) == 1:
+                # only one media kind given, no point showing the dropdown
+                self.fields["kind"].widget = forms.HiddenInput()
+            else:
```
Comment (akx) [inline shoop/admin/modules/products/views/forms.py]: You should filter the choices even if the input is hidden; `choices` affects the validation as well.

File: shoop/admin/modules/products/views/forms.py [hunk 3]
```diff
@@ -178,10 +178,19 @@ def __init__(self, **kwargs):
...
+                # multiple media kinds allowed, filter the choices list to reflect the `self.allowed_media_kinds`
+                allowed_kinds_values = set(v.value for v in self.allowed_media_kinds)
+                self.fields["kind"].choices = [
+                    (value, choice)
+                    for value, choice in self.fields["kind"].choices
+                    if value in allowed_kinds_values
+                ]
```
Comment (akx) [inline shoop/admin/modules/products/views/forms.py]: Does this end up filtering the actual select widget as well?

**Comment by akx** [comment]
:shipit:

**Pull request merged**